packages/dds/tree/src/test/simple-tree/discrepancies.spec.ts

```diff
@@ -0,0 +1,909 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "node:assert";
+import {
+	LeafNodeStoredSchema,
+	MapNodeStoredSchema,
+	ObjectNodeStoredSchema,
+	storedEmptyFieldSchema,
+	ValueSchema,
+	type TreeFieldStoredSchema,
+	type TreeNodeSchemaIdentifier,
+	type TreeStoredSchema,
+} from "../../core/index.js";
+import {
+	defaultSchemaPolicy,
+	allowsRepoSuperset,
+	FieldKinds,
+	type FlexFieldKind,
+} from "../../feature-libraries/index.js";
+import { brand } from "../../util/index.js";
+import {
+	FieldKind,
+	normalizeFieldSchema,
+	SchemaFactoryAlpha,
+	toStoredSchema,
+	isRepoSuperset,
+	getAllowedContentDiscrepancies,
+	type AnnotatedAllowedType,
+	type TreeNodeSchema,
+} from "../../simple-tree/index.js";
+// eslint-disable-next-line import/no-internal-modules
+import { createFieldSchema } from "../../simple-tree/fieldSchema.js";
+// eslint-disable-next-line import/no-internal-modules
+import { fieldSchema } from "../feature-libraries/modular-schema/comparison.spec.js";
+// eslint-disable-next-line import/no-internal-modules
+import { LeafNodeSchema } from "../../simple-tree/leafNodeSchema.js";
+// eslint-disable-next-line import/no-internal-modules
+import { findExtraAllowedTypes } from "../../simple-tree/discrepancies.js";
+
+// Arbitrary schema name used in tests
+const testTreeNodeIdentifier = brand<TreeNodeSchemaIdentifier>("tree");
+const testTreeNodeIdentifierNormalized = brand<TreeNodeSchemaIdentifier>(
+	"schema discrepancies.tree",
+);
+
+const schemaFactory = new SchemaFactoryAlpha("schema discrepancies");
+const numberName = brand<TreeNodeSchemaIdentifier>(schemaFactory.number.identifier);
+
+// TODO:#43720: Add tests for RecordNodeObjects
+describe("Schema Discrepancies", () => {
+	it("Node kind difference for each possible combination", () => {
+		const objectNodeSchema = schemaFactory.objectAlpha(testTreeNodeIdentifier, {
+			x: schemaFactory.required([schemaFactory.number]),
+		});
+		const objectNodeStoredSchema = toStoredSchema(objectNodeSchema);
+
+		const mapNodeSchema = schemaFactory.mapAlpha(testTreeNodeIdentifier, [
+			schemaFactory.number,
+		]);
```

Comment:
[] can be implied, no need to include it

```suggestion
		const mapNodeSchema = schemaFactory.mapAlpha(testTreeNodeIdentifier, SchemaFactory.number);
```